src/mlpack/core/data/save_impl.hpp

```diff
@@ -199,7 +198,7 @@ bool Save(const std::string& filename,
   arma::file_type saveType;
   std::string stringType;
 
-  if (extension == "txt" || extension == "tsv")
+  if (extension == "json" || extension == "tsv")
```

Comment:
```suggestion
  if (extension == "txt" || extension == "tsv")
```
This is in the function that saves an `SpMat`, so that doesn't use cereal, so we can leave it as `txt`.